internal/ansible/runner/runner.go

```diff
@@ -70,11 +70,12 @@ func ansibleVerbosityString(verbosity int) string {
 
 type cmdFuncType func(ident, inputDirPath string, maxArtifacts, verbosity int) *exec.Cmd
 
-func playbookCmdFunc(path string) cmdFuncType {
+func playbookCmdFunc(playbookPath, rolesPath string) cmdFuncType {
 	return func(ident, inputDirPath string, maxArtifacts, verbosity int) *exec.Cmd {
 		cmdOptions := []string{
 			"--rotate-artifacts", fmt.Sprintf("%v", maxArtifacts),
-			"-p", path,
+			"--roles-path", rolesPath,
```

Comment:
Looks like if we pass this arg, we clobber `ANSIBLE_ROLES_PATH` https://github.com/ansible/ansible-runner/blob/0243188274b9debffe141bd6046db619641bc4c7/ansible_runner/utils.py#L210

I think we need to find another way.

Reply:
It doesn't quite get clobbered, just [updated](https://github.com/ansible/ansible-runner/blob/0243188274b9debffe141bd6046db619641bc4c7/ansible_runner/utils.py#L208) with the path `Watch.Role` is set to (effectively `--roles-path={{.Watch.Role}}`). If `ANSIBLE_ROLES_PATH` is set, ansible-operator [sets `w.Role` to that value](https://github.com/operator-framework/operator-sdk/blob/v1.1.0/internal/ansible/watches/watches.go#L262-L270) with a [greater priority](https://github.com/operator-framework/operator-sdk/blob/v1.1.0/internal/ansible/watches/watches.go#L208-L214) than the default `$PWD/roles`; if not set, then it prioritizes the default path. So ultimately `ANSIBLE_ROLES_PATH` is still respected. Does that logic make sense or am I wrong somehow?

Reply:
I'd have to do a more careful review to be sure, but to me it looks like the "roles_path" variable is appended to, but at no point does it check if there is already the `ANSIBLE_ROLES_PATH` environment variable before assigning that value. 

So, it appears that the order of precedence is that it `--roles-path` is passed, it overrides anything in the environment variable.

Reply:
I see what you mean though, if `ANSIBLE_ROLES_PATH` is set, then that will affect the `roles-path` we pass along to ansible-runner. But we won't pass along the entire value of the env var, just one of the paths. https://github.com/operator-framework/operator-sdk/blob/v1.1.0/internal/ansible/watches/watches.go#L210-L212

I can't think of a reason why operator authors would want to store roles in more than one roles directory, but since that is a supported case with Ansible itself I think we need to allow it.